core/src/utils/test/playwright.ts

```diff
@@ -14,7 +14,19 @@ export const test = base.extend({
      */
     page.goto = (url: string) => {
       const { mode, rtl } = testInfo.project.metadata;
-      const formattedUrl = `${url}?ionic:_testing=true&ionic:mode=${mode}&rtl=${rtl}`;
+
+      const splitUrl = url.split('?');
+      const paramsString = splitUrl[1];
```

Comment:
Will `url` always have params in it already? (Wondering if directly hitting `splitUrl[1]` with no safeguards will cause problems. Also applies to `getSnapshotSettings`)

Reply:
It will only have params in it if you are explicitly overriding the mode/RTL config. If you index into an out of bounds item in the array with JS, you get `undefined` instead of an error.

The `init` parameter for `URLSearchParams` is optional, so passing in `undefined` will not error either: https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/URLSearchParams